### PR TITLE
chore: small fixes around tiering

### PR DESCRIPTION
**Pull request by romange**
There are no changes in functionality here.

### CODE REVIEW

File: src/core/compact_object.cc [hunk 1]
```diff
@@ -950,21 +950,23 @@ void CompactObj::SetExternal(size_t offset, uint32_t sz) {
   u_.ext_ptr.offload.page_index = offset / 4096;
 }
 
-void CompactObj::SetCold(size_t offset, uint32_t sz, detail::TieredColdRecord* record) {
-  SetMeta(EXTERNAL_TAG, mask_);
+void CompactObj::SetCool(size_t offset, uint32_t sz, detail::TieredColdRecord* record) {
```
Comment (romange) [inline src/core/compact_object.cc:953]: changed naming from Cold to Cool

File: src/core/compact_object.cc [hunk 2]
```diff
@@ -950,21 +950,23 @@ void CompactObj::SetExternal(size_t offset, uint32_t sz) {
...
+  // We copy the mask of the "cooled" referenced object because it contains the encoding info.
+  SetMeta(EXTERNAL_TAG, record->value.mask_);
+
   u_.ext_ptr.is_cool = 1;
   u_.ext_ptr.page_offset = offset % 4096;
   u_.ext_ptr.serialized_size = sz;
-  u_.ext_ptr.cold_record = record;
+  u_.ext_ptr.cool_record = record;
```
Comment (romange) [inline src/core/compact_object.cc:960]: same

File: src/core/compact_object.cc [hunk 3]
```diff
@@ -976,9 +978,9 @@ void CompactObj::ImportExternal(const CompactObj& src) {
 
 std::pair<size_t, size_t> CompactObj::GetExternalSlice() const {
   DCHECK_EQ(EXTERNAL_TAG, taglen_);
-  DCHECK_EQ(u_.ext_ptr.is_cool, 0);
-
-  size_t offset = size_t(u_.ext_ptr.offload.page_index) * 4096 + u_.ext_ptr.page_offset;
+  auto& ext = u_.ext_ptr;
```
Comment (romange) [inline src/core/compact_object.cc:981]: support cool and offline objects.

**Review by kostasrim — APPROVED** [review]
cool

**Pull request merged**